MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/AddressServiceDefault.java

```diff
@@ -77,6 +81,33 @@ public CancellableTask createAddress(final Long customerId, final Address addres
             .observeOn(callbackScheduler);
     }
 
+    public CancellableTask deleteAddress(Long customerId, Long addressId, Callback<Void> callback) {
+        return new CancellableTaskSubscriptionWrapper(deleteAddress(customerId, addressId).subscribe(new InternalCallbackSubscriber<>(callback)));
+    }
+
+    public Observable<Void> deleteAddress(Long customerId, Long addressId) {
+        if (customerId == null) {
+            throw new NullPointerException("customerId cannot be null");
+        }
+
+        if (addressId == null) {
+            throw new NullPointerException("addressId cannot be null");
+        }
+
+        int[] successCodes = {HTTP_NO_CONTENT};
+
+        return retrofitService.deleteAddress(customerId, addressId)
+            .doOnNext(new RetrofitSuccessHttpStatusCodeHandler<>(successCodes))
+            .map(new Func1<Response<Void>, Void>() {
```

Comment:
I thinks so - we expect to return a Void but what we get back from retrofit is a Response<Void>.
Is there a better way to do this?